public/javascripts/components/appCardsAndCategory.js

```diff
@@ -0,0 +1,95 @@
+app.component('homepage-base', {
+  data() {
+    return {
+      apps: [],
+      appsIsEmpty: false,
+      categories: []
+    }
+  },
+
+  created: function () {
+    fetch(`/api/categories`)
+      .then(response => response.json())
+      .then(res => {
+        this.categories = res.data.categories;
+      }).then(() => {
+        fetch(`/api/category/all`)
```

Comment:
`api/categories` is for the categories objects only.. like it'll return all our categories info.
`api/category/all` is like `api/category/food` but instead of just apps of food, it'll return all apps. Ly has to make some tweaks for the Vue that's why we have this extra api endpoints just to return all apps